homeassistant/components/netatmo/binary_sensor.py

```diff
@@ -35,130 +34,61 @@
-                            module_name,
-                            home,
-                            timeout,
-                            camera_type,
-                            variable,
-                        )
-                    ],
-                    True,
-                )
+    def get_devices():
+        """Retrieve Netatmo devices."""
+        devices = []
+
+        try:
+            data = CameraData(hass, auth)
+        except NoDevice:
+            _LOGGER.debug("No camera devices to add")
+
+        for camera_id in data.get_all_camera_ids():
+            camera_type = data.get_camera_type(camera_id=camera_id)
+            home_id = data.get_camera_home_id(camera_id=camera_id)
 
-        for module_name in data.get_module_names(camera_name):
-            for variable in tag_sensors:
-                camera_type = None
-                add_entities(
-                    [
-                        NetatmoBinarySensor(
-                            data,
-                            camera_name,
-                            module_name,
-                            home,
-                            timeout,
-                            camera_type,
-                            variable,
-                        )
-                    ],
-                    True,
+            for sensor_name in SENSOR_TYPES[camera_type]:
+                devices.append(
+                    NetatmoBinarySensor(data, camera_id, home_id, sensor_name)
                 )
 
+        return devices
+
+    async_add_entities(await hass.async_add_executor_job(get_devices), True)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the access to Netatmo binary sensor."""
+    return
+
 
 class NetatmoBinarySensor(BinarySensorDevice):
     """Represent a single binary sensor in a Netatmo Camera device."""
 
-    def __init__(
-        self, data, camera_name, module_name, home, timeout, camera_type, sensor
-    ):
+    def __init__(self, data, camera_id, home_id, sensor_type, module_id=None):
         """Set up for access to the Netatmo camera events."""
         self._data = data
-        self._camera_name = camera_name
-        self._module_name = module_name
-        self._home = home
-        self._timeout = timeout
-        if home:
-            self._name = f"{home} / {camera_name}"
-        else:
-            self._name = camera_name
-        if module_name:
-            self._name += " / " + module_name
-        self._sensor_name = sensor
-        self._name += " " + sensor
-        self._cameratype = camera_type
+        self._camera_id = camera_id
+        self._module_id = module_id
+        self._sensor_type = sensor_type
+        camera_info = data.camera_data.cameraById(cid=camera_id)
+        self._camera_name = camera_info["name"]
+        self._camera_type = camera_info["type"]
+        if module_id:
+            self._module_name = data.camera_data.moduleById(mid=module_id)["name"]
```

Comment:
same.

Reply:
No, it just retrieves the info from a cached dictionary.